core/domain/draft_upgrade_services.py

```diff
@@ -312,6 +312,91 @@ def _convert_html_in_draft_change_list(
                 })
         return draft_change_list
 
+    @classmethod
+    def _convert_states_v52_dict_to_v53_dict(
+        cls, draft_change_list: List[exp_domain.ExplorationChange]
+    ) -> List[exp_domain.ExplorationChange]:
+        """Converts from version 52 to 53. Version 53 fixes general
+        state, interaction and rte data. This will update the drafts
+        for state and RTE part but won't be able to do for interaction.
+        The `ExplorationChange` object for interaction is divided into
+        further properties and we won't be able to collect enough info
+        to update the draft.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+
+        Raises:
+            InvalidDraftConversionException. The conversion cannot be
+                completed.
+        """
+        for exp_change in draft_change_list:
+            if exp_change.cmd != exp_domain.CMD_EDIT_STATE_PROPERTY:
```

Comment:
It does not cover the cases for `interactions` the reason being is that we will not be able to collect enough info about the interactions in order to update it as the interactions are divided into further properties when it comes in the form of change list.
By this, I mean that we will require a complete interaction in order to make the appropriate changes. Like [here](https://github.com/oppia/oppia/blob/develop/core/domain/exp_domain.py#L292) the interactions are divided into further properties and there is a strong possibility that we will not be having exploration change for each interaction property and that is why we will not be editing it.
I was also talking this part with @vojtechjelinek and he agrees.
Thanks

Reply:
In those cases should we not do the conversion and mark the draft invalid? In general, I think this migration is big and any state_property changes should lead to an invalid draft. This will help us avoid having invalid data in the draft which get stuck in the validation pipeline!